lib/azure/armrest/storage_account_service.rb

```diff
@@ -261,7 +261,12 @@ def get_private_images(storage_accounts)
 
         storage_accounts.each do |lstorage_account|
           threads << Thread.new(lstorage_account) do |storage_account|
-            key = get_account_key(storage_account)
+            begin
+              key = get_account_key(storage_account)
+            rescue Azure::Armrest::ApiException => err
+              next
```

Comment:
Let's add a comment here after "next" that says, "Most likely because account isn't completely provisioned yet" or something.